From 582cbde53cfb9e79182663bfd7c86aa46c0b46eb Mon Sep 17 00:00:00 2001
From: Kristen McWilliam <kmcwilliampublic@gmail.com>
Date: Thu, 29 Aug 2024 15:20:54 -0400
Subject: [PATCH 1/3] kcms: pull dmidecode-helper into a separate dir

This allows us to reuse the helper in other
places, like the memory info kcm.
---
 kcms/CMakeLists.txt                                           | 1 +
 kcms/about-distro/src/CMakeLists.txt                          | 2 --
 kcms/helpers/CMakeLists.txt                                   | 4 ++++
 .../src => helpers}/dmidecode-helper/CMakeLists.txt           | 0
 .../{about-distro/src => helpers}/dmidecode-helper/helper.cpp | 0
 kcms/{about-distro/src => helpers}/dmidecode-helper/helper.h  | 0
 .../dmidecode-helper/org.kde.kinfocenter.dmidecode.actions    | 0
 7 files changed, 5 insertions(+), 2 deletions(-)
 create mode 100644 kcms/helpers/CMakeLists.txt
 rename kcms/{about-distro/src => helpers}/dmidecode-helper/CMakeLists.txt (100%)
 rename kcms/{about-distro/src => helpers}/dmidecode-helper/helper.cpp (100%)
 rename kcms/{about-distro/src => helpers}/dmidecode-helper/helper.h (100%)
 rename kcms/{about-distro/src => helpers}/dmidecode-helper/org.kde.kinfocenter.dmidecode.actions (100%)

diff --git a/kcms/CMakeLists.txt b/kcms/CMakeLists.txt
index f283f477..ce534f40 100644
--- a/kcms/CMakeLists.txt
+++ b/kcms/CMakeLists.txt
@@ -1,3 +1,4 @@
+add_subdirectory(helpers)
 
 add_subdirectory( about-distro )
 add_subdirectory(vulkan)
diff --git a/kcms/about-distro/src/CMakeLists.txt b/kcms/about-distro/src/CMakeLists.txt
index fffb24d6..961ac959 100644
--- a/kcms/about-distro/src/CMakeLists.txt
+++ b/kcms/about-distro/src/CMakeLists.txt
@@ -1,8 +1,6 @@
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/Version.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/Version.h)
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
-add_subdirectory(dmidecode-helper)
-
 set(kcm_SRCS
     CPUEntry.cpp
     CPUEntry.h
diff --git a/kcms/helpers/CMakeLists.txt b/kcms/helpers/CMakeLists.txt
new file mode 100644
index 00000000..cbf024bd
--- /dev/null
+++ b/kcms/helpers/CMakeLists.txt
@@ -0,0 +1,4 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# SPDX-FileCopyrightText: 2024 Kristen McWilliam <kmcwilliampublic@gmail.com>
+
+add_subdirectory(dmidecode-helper)
diff --git a/kcms/about-distro/src/dmidecode-helper/CMakeLists.txt b/kcms/helpers/dmidecode-helper/CMakeLists.txt
similarity index 100%
rename from kcms/about-distro/src/dmidecode-helper/CMakeLists.txt
rename to kcms/helpers/dmidecode-helper/CMakeLists.txt
diff --git a/kcms/about-distro/src/dmidecode-helper/helper.cpp b/kcms/helpers/dmidecode-helper/helper.cpp
similarity index 100%
rename from kcms/about-distro/src/dmidecode-helper/helper.cpp
rename to kcms/helpers/dmidecode-helper/helper.cpp
diff --git a/kcms/about-distro/src/dmidecode-helper/helper.h b/kcms/helpers/dmidecode-helper/helper.h
similarity index 100%
rename from kcms/about-distro/src/dmidecode-helper/helper.h
rename to kcms/helpers/dmidecode-helper/helper.h
diff --git a/kcms/about-distro/src/dmidecode-helper/org.kde.kinfocenter.dmidecode.actions b/kcms/helpers/dmidecode-helper/org.kde.kinfocenter.dmidecode.actions
similarity index 100%
rename from kcms/about-distro/src/dmidecode-helper/org.kde.kinfocenter.dmidecode.actions
rename to kcms/helpers/dmidecode-helper/org.kde.kinfocenter.dmidecode.actions
-- 
2.47.0

